update-versions.py

```diff
@@ -103,16 +104,27 @@ def nix_prefetch(args):
 g = github.Github(auth=auth)
 repo = g.get_repo("hashicorp/terraform")
 # TODO: Drop "v" prefix first
-# TODO: Combine the output of the following endpoints to get a list of active
-# versions
-# https://api.github.com/repos/hashicorp/terraform/tags
-# https://endoflife.date/api/terraform.json
-releases = list(filter(is_stable, repo.get_releases()))
+
+# Craft list of maintaned releases by filtering the release list with the EOL data
+eol_data = requests.get("https://endoflife.date/api/terraform.json").json()
+maintained_cycles = [cycle['cycle'] for cycle in eol_data if not cycle['eol']]
+maintained_releases = list(filter(lambda version: is_stable(version, maintained_cycles), repo.get_releases()))
```

Comment:
@oscar-izval what are your thoughts about wrapping lines 109-110 in a function to make it more clear what this section does?

```py
def get_maintained_cycles():
  eol_data = requests.get("https://endoflife.date/api/terraform.json").json()
  return [cycle['cycle'] for cycle in eol_data if not cycle['eol']]
```

Reply:
Sounds good to me! 🙇‍♂️